pkg/metrics/builder/builder_test.go

```diff
@@ -0,0 +1,226 @@
+package builder
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+func getDescStruct(opts MetricOpts) string {
+	var labels []string
+	lpStrings := make([]string, 0, len(opts.labelKeyValues))
+	for key, value := range opts.labelKeyValues {
+		lpStrings = append(lpStrings, fmt.Sprintf("%s=%q", key, value))
+	}
+	sort.Strings(lpStrings)
+
+	return fmt.Sprintf("Desc{fqName: %q, help: %q, constLabels: {%s}, variableLabels: %v}",
+		opts.name, opts.desc, strings.Join(lpStrings, ","), labels)
```

Comment:
no assignment for labels in this function.

Reply:
This is actually intended as we don't use variable labels and it's specific to Prometheus as to how the text appears. The motivation behind testing that was I wanted to see if the Collector object actually has the values that were in the MetricBuilder and all the Collector's properties are private except a String. It's a roundabout way to test if the values are being set but I think it's the only way as all are private.